ext/mysql2/client.c

```diff
@@ -539,21 +540,35 @@ static VALUE disconnect_and_raise(VALUE self, VALUE error) {
   return Qnil;
 }
 
-static VALUE do_query(void *args) {
-  struct async_query_args *async_args;
-  struct timeval tv;
-  struct timeval* tvp;
-  long int sec;
+static void wait_for_fd(int fd, struct timeval *tvp) {
   int retval;
+
+  for(;;) {
+    retval = rb_wait_for_single_fd(fd, RB_WAITFD_IN, tvp);
```

Comment:
I was surprised, so I double-checked that `rb_wait_for_single_fd` does allow a NULL timeval argument. Could you add a comment to let the future coders know that NULL can get through from `get_read_timeout` and is a legal value for this function?

https://github.com/ruby/ruby/search?utf8=%E2%9C%93&q=rb_wait_for_single_fd